nixos/tests/amazon-image.nix

```diff
@@ -0,0 +1,60 @@
+# Tests building and running a GUID Partition Table (GPT) appliance image.
+# "Appliance" here means that the image does not contain the normal NixOS
+# infrastructure of a system profile and cannot be re-built via
+# `nixos-rebuild`.
+
+{ lib, ... }:
+
+let rootPartitionLabel = "root";
+in {
+  name = "appliance-gpt-image";
+
+  meta.maintainers = with lib.maintainers; [ arianvp ];
+
+  defaults = { lib, ... }: {
+    imports = [ ../modules/image/amazon.nix ];
+
+    virtualisation.directBoot.enable = false;
+    virtualisation.mountHostNixStore = false;
+    virtualisation.installBootLoader = true;
+
+    virtualisation.fileSystems = lib.mkForce {
+      "/" = {
+        device = "/dev/disk/by-partlabel/root";
+        fsType = "ext4";
+      };
+    };
+
+    image.repart = {
+      # OVMF does not work with the default repart sector size of 4096
+      sectorSize = 512;
```

Comment:
Maybe we should default this in the image? I expect many people booting using OVMF.